CountingBloomFilter.hpp

```diff
@@ -271,61 +252,76 @@ template<typename T>
 void
 CountingBloomFilter<T>::readFilter(const std::string& path)
 {
-	FILE* file;
-	if ((file = fopen(path.c_str(), "rb")) == nullptr) {
+	std::ifstream file(path);
+	if (!file.is_open()) {
 		std::cerr << "ERROR: Failed to open file: " << path << "\n";
 		exit(EXIT_FAILURE);
 	}
 	readHeader(file);
-	struct stat buf;
-	if (fstat(fileno(file), &buf) != 0) {
-		std::cerr << "ERROR: Failed to open file: " << path << "\n";
-		exit(EXIT_FAILURE);
-	}
-	size_t arraySizeOnDisk = buf.st_size - sizeof(struct FileHeader);
-	if (arraySizeOnDisk != m_sizeInBytes) {
-		std::cerr << "ERROR: File size of " << path << " (" << arraySizeOnDisk << " bytes), "
-		          << "does not match size read from its header (" << m_sizeInBytes << " bytes).\n";
-		exit(EXIT_FAILURE);
-	}
-
-	size_t nread = fread(m_filter, arraySizeOnDisk, 1, file);
-	if (nread != 1 && fclose(file) != 0) {
+	char* filter;
+	filter = new char[m_sizeInBytes];
```

Comment:
```suggestion
	std::vector<uint8_t> filter([m_sizeInBytes]);
```
Use a `std::vector` of `uint8_t` rather than `char`. Only use `char` for ASCII or UTF-8 strings, not for binary data.